File: types/go/CreatorStruct.go
package main

type Creator	 struct {
	Aliases []string `json:"aliases"`
	Discord *Discord `json:"discord"`
	Name    string   `json:"name"`
	Steam   *Steam   `json:"steam"`
	Twitch  *Twitch  `json:"twitch"`
	Youtube *Youtube `json:"youtube"`
}

type Discord struct {
	ID       *string `json:"id"`
	Partner  *bool   `json:"partner"`
	Username *string `json:"username"`
}

type Steam struct {
	ID   *string `json:"id"`
	Name *string `json:"name"`
}

type Twitch struct {
	ID      *string `json:"id"`
	Partner *bool   `json:"partner"`
}

type Youtube struct {
	Handle  *string `json:"handle"`
	ID      *string `json:"id"`
	Partner *bool   `json:"partner"`
}

File: types/rust/CreatorStruct.rs
use serde::{Serialize, Deserialize};

#[derive(Serialize, Deserialize)]
pub struct CreatorStruct {
    aliases: Option<Vec<String>>,
    discord: Option<Discord>,
    name: String,
    steam: Option<Steam>,
    twitch: Option<Twitch>,
    youtube: Option<Youtube>,
}

#[derive(Serialize, Deserialize)]
pub struct Discord {
    id: Option<String>,
    partner: Option<bool>,
    username: Option<String>,
}

#[derive(Serialize, Deserialize)]
pub struct Steam {
    id: Option<String>,
    name: Option<String>,
}

#[derive(Serialize, Deserialize)]
pub struct Twitch {
    id: Option<String>,
    partner: Option<bool>,
}

#[derive(Serialize, Deserialize)]
pub struct Youtube {
    handle: Option<String>,
    id: Option<String>,
    partner: Option<bool>,
}
